docs/sources/flow/concepts/modules.md

```diff
@@ -0,0 +1,145 @@
+---
+aliases:
+- ../../concepts/modules/
+title: Modules
+weight: 300
+---
+
+# Modules
+
+_Modules_ are a way to create Grafana Agent Flow configuration files which can
+laoded as a component. Modules are a great way to parameterize configuration
+files to create reusable pipelines.
+
+Modules are Grafana Agent Flow configuration files which have:
+
+* Arguments: settings which configure a module.
+* Exports: named values which a module exposes to the consumer of the module.
+* Components: Grafana Agent Flow Components to run when the module is running.
+
+Modules are loaded into Grafana Agent Flow by using a [Module
+loader](#module-loaders).
+
+Refer to the documentation for the [argument block][] and [export block][] to
+learn how to define arguments and exports for a module.
+
+[argument block]: {{< relref "../reference/config-blocks/argument.md" >}}
+[export block]: {{< relref "../reference/config-blocks/export.md" >}}
+
+## Module loaders
+
+A _Module loader_ is a Grafana Agent Flow component which retreives a module
+and runs the components defined inside of it.
+
+Module loader components are responsible for:
+
+* Retreiving the module source to run.
+* Creating a [Component controller][] for the module to run in.
+* Passing arguments to the loaded module.
+* Exposing exports from the loaded module.
+
+Module loaders typically are called `module.LOADER_NAME`. The list of module
+loader components can be found in the list of Grafana Agent Flow
+[Components][].
+
+Some module loaders may not support running modules with arguments or exports;
+refer to the documentation for the module loader you are using for more
+information.
+
+[Component controller]: {{< relref "./component_controller.md" >}}
+[Component]: {{< relref "../reference/components/" >}}
+
+## Module sources
```

Comment:
are we going with `source` over `content`? I don't have an opinion but think we should update the rfc if going with `source`.

Reply:
I'll change this to sync up with whatever the merged component uses, I just went with what's in the PR for now